umbraco/Umbraco-CMS
# Improvements to the OpenAPI Docs

### Description
This PR ensures that the endpoints in the management API, reports it can return the following statuses

- **401 - Unauthorized**
  - If it do not have `AllowAnonymousAttribute` - This is the same logic that already shows the authentication "lock" symbol
- **403 - Forbidden**
  - If it has at least 3 `AuthorizeAttribute`
    - The two from `ManagementApiControllerBase` will always ensure you have access to backoffice and can disable the controller. If other are present, there is a risk the user do not have access, and thereby a 403 returned.
  - If the constructor injects `IAuthorizationService`, as all dynamic policies will use this pattern.  Also here there is a risk the user do not have access to a given resource and thereby a 403 returned.
  
 
Furthermore, we also ensure the APIs only accept the `application/json` header, and not `plain/text` and other defaults.


### Test
- Ensure endpoints that require authentication documents the possible 401
- Ensure endpoints that require but potentially your user do not have access to, documents the possible 403


### Bonus
I also tested what happens if I am authenticated but I remove all the tokens from the database. In that case a 401 is returned as expected.
